git-webui/release/share/git-webui/webui/js/git-webui.js

```diff
@@ -1937,6 +1947,108 @@ webui.DiffView = function(sideBySide, hunkSelectionAllowed, parent, stashedCommi
     var gitApplyType = "stage";
 };
 
+webui.DiscardedView = function(mainView) {
+    var self = this;
+
+    self.show = function() {
+        self.update();
+        mainView.switchTo(self.element);
+    };
+
+
+    self.update = function() {
+        self.discarded = [];
+        discardedList.innerHTML = '';
+        $('.file-contents').html('');
+        $('.contents-menu').removeClass("has-items");
+        $('.restore-discarded').html('');
+        $('.external-name').text('');
+        
+        $.get("discarded-states", function(discarded) {
+            self.discarded = JSON.parse(discarded);
+            if (self.discarded.length == 0) {
+                discardedList.innerHTML = '<h4>You have no saved discarded states</h4>';
+            }
+            
+            self.populateUiWithDiscardedStates();
+        });
+
+        
+    }
+
+    self.populateUiWithDiscardedStates = function() {
+        self.discarded.forEach((discardedState, ind) => {
+            var discardedListEntry = $(
+                '<a class="log-entry list-group-item">' + 
+                    '<header>' + 
+                        '<h6 class="file-internalname">' + discardedState.Name + '</h6>' + 
+                        '<span class="discard-date">' + self.formatTimestamp(discardedState.Timestamp) + '</span>' +
+                    '</header>' +
+                    '<span> Branch: ' + discardedState.Branch + ' </span> <br>' +
+                    '<span> User: '+ discardedState.Username + '</span>' +
+
+                '</a>'
+            );
+            
+            discardedListEntry.on("click", function() {
+                $('.log-entry.list-group-item').removeClass('active');
+                $(this).addClass('active');
+
+                $('.contents-menu').addClass("has-items");
+                $('.external-name').text(self.discarded[ind].FullExternalName);
+                
+                var btn = $('<button class="btn btn-sm btn-primary restore-discarded-btn">Restore</button>')[0];
+                var discardElement = $('.restore-discarded')[0];
+                discardElement.innerHTML = '';
+                discardElement.appendChild(btn);
+
+                $('.restore-discarded-btn').off("click").on("click", function() {
+                    $.post("restore-discarded", {file: discardedState.Id}, function(message) {
+                        if (message.trim() == "Please commit changes to file before restoring discarded state") {
+                            webui.showError(message);
+                        } else {
+                            webui.showSuccess(message);
+                        }
+                        
+                        self.update();
+                    });
+                });
+
+                $('.file-contents').html('<pre>' + discardedState.Contents + '</pre>');
```

Comment:
We need to escape HTML tags in discardedState.Contents. As is, XData blocks like a production definition will not display correctly:
![image](https://github.com/user-attachments/assets/751f6fc2-b42b-40b3-b13c-7aee09869ea8)